bdsg/include/bdsg/snarl_distance_index.hpp

```diff
@@ -813,11 +813,18 @@ class SnarlDistanceIndex : public SnarlDecomposition, public TriviallySerializab
      *   The root vector stores the root of every connected component, which can be a 
      *   node, snarl, or chain
      */
-    const static size_t ROOT_RECORD_SIZE = 5;
-    const static size_t COMPONENT_COUNT_OFFSET = 1;
-    const static size_t NODE_COUNT_OFFSET = 2;
-    const static size_t MIN_NODE_ID_OFFSET = 3;
-    const static size_t MAX_TREE_DEPTH_OFFSET = 4;
+    const static size_t ROOT_RECORD_SIZE = 6;
+    const static size_t VERSION_NUMBER_OFFSET = 1;
+    const static size_t COMPONENT_COUNT_OFFSET = 2;
+    const static size_t NODE_COUNT_OFFSET = 3;
+    const static size_t MIN_NODE_ID_OFFSET = 4;
+    const static size_t MAX_TREE_DEPTH_OFFSET = 5;
+
+    // While the version number is 3, we will store (max - version)
+    // to avoid getting confused with old indexes without version numbers
+    // that start with component count
+    const static size_t CURRENT_VERSION_NUMBER = 3;
+    const static size_t VERSION_NUMBER_SENTINEL = (1 << 26) - CURRENT_VERSION_NUMBER;
```

Comment:
Setting some high bits will make for a big page in the paged vector, but that shouldn't be a big problem since it's just the first page.

Why 26? And why subtract from it instead of setting a high bit and masking down to the bits where the version actually is? I think this way we can keep our values at the high end in 1 fewer bits than with the masking approach, which then means 1 fewer bits in each entry in the page, which is nice. But if that's the reasoning it should probably be explained.